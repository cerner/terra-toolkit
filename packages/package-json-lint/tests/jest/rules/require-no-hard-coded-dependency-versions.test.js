const requireNoHardCodedDependencyVersions = require('../../../src/rules/require-no-hard-coded-dependency-versions');

describe('require-no-hard-coded-dependency-versions', () => {
  describe('when projectType is application', () => {
    it('succeeds when there are hardcoded dependencies', () => {
      let results;
      requireNoHardCodedDependencyVersions.create({
        ruleConfig: {
          severity: {
            severityType: 'error',
          },
        },
        projectType: 'application',
        report: (issues) => {
          results = issues;
        },
      }).dependencies({
        a: '1.0.0',
      });
      expect(results).toMatchSnapshot();
    });
    it('succeeds when there are no hardcoded dependencies', () => {
      let results;
      requireNoHardCodedDependencyVersions.create({
        ruleConfig: {
          severity: {
            severityType: 'error',
          },
        },
        projectType: 'application',
        report: (issues) => {
          results = issues;
        },
      }).dependencies({
        a: '^1.0.0',
      });
      expect(results).toMatchSnapshot();
    });
  });

  describe('when projectType is module', () => {
    it('fails when there are hardcoded dependencies', () => {
      let results;
      requireNoHardCodedDependencyVersions.create({
        ruleConfig: {
          severity: {
            severityType: 'error',
          },
        },
        projectType: 'module',
        report: (issues) => {
          results = issues;
        },
      }).dependencies({
        a: '1.0.0',
      });
      expect(results).toMatchSnapshot();
    });
    it('succeeds when there are no hardcoded dependencies', () => {
      let results;
      requireNoHardCodedDependencyVersions.create({
        ruleConfig: {
          severity: {
            severityType: 'error',
          },
        },
        projectType: 'module',
        report: (issues) => {
          results = issues;
        },
      }).dependencies({
        a: '^1.0.0',
        b: '1.0.0 - 2.9999.9999',
        c: '>=1.0.2 <2.1.2',
        d: '>1.0.2 <=2.3.4',
        e: '<1.0.0 || >=2.3.1 <2.4.5 || >=2.5.2 <3.0.0',
        f: '~1.2',
        g: 'latest',
        h: '~1.2.3',
        i: '2.x',
        j: '3.3.x',
      });
      expect(results).toMatchSnapshot();
    });
  });

  describe('when projectType is devModule', () => {
    it('fails when there are hardcoded dependencies', () => {
      let results;
      requireNoHardCodedDependencyVersions.create({
        ruleConfig: {
          severity: {
            severityType: 'error',
          },
        },
        projectType: 'devModule',
        report: (issues) => {
          results = issues;
        },
      }).dependencies({
        a: '1.0.0',
      });
      expect(results).toMatchSnapshot();
    });
    it('succeeds when there are no hardcoded dependencies', () => {
      let results;
      requireNoHardCodedDependencyVersions.create({
        ruleConfig: {
          severity: {
            severityType: 'error',
          },
        },
        projectType: 'devModule',
        report: (issues) => {
          results = issues;
        },
      }).dependencies({
        a: '^1.0.0',
        b: '1.0.0 - 2.9999.9999',
        c: '>=1.0.2 <2.1.2',
        d: '>1.0.2 <=2.3.4',
        e: '<1.0.0 || >=2.3.1 <2.4.5 || >=2.5.2 <3.0.0',
        f: '~1.2',
        g: 'latest',
        h: '~1.2.3',
        i: '2.x',
        j: '3.3.x',
      });
      expect(results).toMatchSnapshot();
    });
  });

  it('succeeds when hardcoded dependency is passed in the allowList', () => {
    let results;
    requireNoHardCodedDependencyVersions.create({
      ruleConfig: {
        severity: {
          severityType: 'error',
          allowList: ['a'],
        },
      },
      projectType: 'devModule',
      report: (issues) => {
        results = issues;
      },
    }).dependencies({
      a: '1.0.0',
    });
    expect(results).toMatchSnapshot();
  });
});
