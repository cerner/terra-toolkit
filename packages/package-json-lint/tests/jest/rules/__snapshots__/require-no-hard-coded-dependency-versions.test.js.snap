// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require-no-hard-coded-dependency-versions succeeds when hardcoded dependency is passed in the allowList 1`] = `undefined`;

exports[`require-no-hard-coded-dependency-versions when projectType is application succeeds when there are hardcoded dependencies 1`] = `undefined`;

exports[`require-no-hard-coded-dependency-versions when projectType is application succeeds when there are no hardcoded dependencies 1`] = `undefined`;

exports[`require-no-hard-coded-dependency-versions when projectType is devModule fails when there are hardcoded dependencies 1`] = `
Object {
  "lintId": "require-no-hard-coded-dependency-versions",
  "lintMessage": "The dependencies for this project have hard-coded versions that violates the require-no-hard-coded-dependency-versions rule:
  a@1.0.0 does not satisfy requirement for the require-no-hard-coded-dependency-versions rule.",
  "projectType": "devModule",
  "severity": Object {
    "severityType": "error",
  },
}
`;

exports[`require-no-hard-coded-dependency-versions when projectType is devModule succeeds when there are no hardcoded dependencies 1`] = `undefined`;

exports[`require-no-hard-coded-dependency-versions when projectType is module fails when there are hardcoded dependencies 1`] = `
Object {
  "lintId": "require-no-hard-coded-dependency-versions",
  "lintMessage": "The dependencies for this project have hard-coded versions that violates the require-no-hard-coded-dependency-versions rule:
  a@1.0.0 does not satisfy requirement for the require-no-hard-coded-dependency-versions rule.",
  "projectType": "module",
  "severity": Object {
    "severityType": "error",
  },
}
`;

exports[`require-no-hard-coded-dependency-versions when projectType is module succeeds when there are no hardcoded dependencies 1`] = `undefined`;
